modules/apps/portal-crypto-hash/portal-crypto-hash-impl/src/test/java/com/liferay/portal/crypto/hash/internal/CryptoHashGeneratorTest.java

```diff
@@ -38,25 +39,31 @@
 
 	@Before
 	public void setUp() throws Exception {
-		CryptoHashProviderRegistry cryptoHashProviderRegistry =
-			new CryptoHashProviderRegistry();
+		CryptoHashProviderFactoryRegistry cryptoHashProviderFactoryRegistry =
+			new CryptoHashProviderFactoryRegistry();
 
-		BCryptCryptoHashProvider bCryptCryptoHashProvider =
-			new BCryptCryptoHashProvider();
+		BCryptCryptoHashProviderFactory bCryptCryptoHashProviderFactory =
+			new BCryptCryptoHashProviderFactory();
 
-		cryptoHashProviderRegistry.register(bCryptCryptoHashProvider);
+		cryptoHashProviderFactoryRegistry.register(
+			bCryptCryptoHashProviderFactory);
 
-		MessageDigestCryptoHashProvider messageDigestCryptoHashProvider =
-			new MessageDigestCryptoHashProvider();
+		MessageDigestCryptoHashProviderFactory
+			messageDigestCryptoHashProviderFactory =
+				new MessageDigestCryptoHashProviderFactory();
 
-		cryptoHashProviderRegistry.register(messageDigestCryptoHashProvider);
+		cryptoHashProviderFactoryRegistry.register(
+			messageDigestCryptoHashProviderFactory);
 
 		_cryptoHashGenerators = Arrays.asList(
-			new CryptoHashGeneratorImpl(bCryptCryptoHashProvider),
-			new CryptoHashGeneratorImpl(messageDigestCryptoHashProvider));
+			new CryptoHashGeneratorImpl(
+				bCryptCryptoHashProviderFactory.create(null)),
```

Comment:
hey... let's add some properties here or, id we are going to allow to pass no properties then we should have a create method with no parameters but it is better to not expose a null

Reply:
Done: https://github.com/liferay-appsec/liferay-portal/pull/294/commits/effcfb04fae8d4595a2237ee22fb20b29a8d7ed5#diff-518be9436a0b9a4268b80d175655cfe08b53f5d82438f1285406fc7f78c4cb23R61